### DISCUSSION TITLE
Can't pull and push images to local nexus in  DAGs

### DISCUSSION THREAD

**Discussion by Powerostad** [apache/airflow#40234] [General] [answered]
Hi
The problem is that I'm trying to create a sub process that will run this code in my DAG:
`docker login -u username -p password insecure_registry_ip`

I have noticed that the airflow has the docker cli but seems like it doesn't contain docker.sock. I have volume from my server docker.sock to the services of my airflow container. 

I don't know if the way I'm trying to use docker in my DAG is correct or not, but I'm seeking help for how to use docker login on a insecure registry in DAGS ? (I'm using main docker-compose of airflow and also extended the airflow image with a docker file)

**Comment by potiuk** [comment]
CAN'T STAND ALL CAPS TITLES IN DISCUSSIONS

**Reply by Powerostad** [reply]
now its better. how about now ?

**Comment by potiuk** [comment, marked as answer]
You need to use docker-in-docker and mount your sock to your image when you run it. That's standard way how you run docker commands from inside docker - and it's a deployment configuration not airflow image configuration, You will google it when searching "docker-in-docker"

**Reply by Powerostad** [reply]
Thank you for your responding
